code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm

```diff
@@ -132,22 +132,13 @@
 		consume_bait()
 
 /mob/living/simple_animal/hostile/asteroid/basilisk/watcher/proc/consume_bait()
-	var/obj/item/stack/ore/diamond/diamonds = locate(/obj/item/stack/ore/diamond) in oview(src, 9)
-	var/obj/item/pen/survival/bait = locate(/obj/item/pen/survival) in oview(src, 9)
-	if(!diamonds && !bait)
-		return
-	if(diamonds)
-		var/distanced = 0
-		distanced = get_dist(loc,diamonds.loc)
-		if(distanced <= 1 && diamonds)
-			qdel(diamonds)
-			src.visible_message(span_notice("[src] consumes [diamonds], and it disappears! ...At least, you think."))
-	if(bait)
-		var/distanceb = 0
-		distanceb = get_dist(loc,bait.loc)
-		if(distanceb <= 1 && bait)
-			qdel(bait)
-			src.visible_message(span_notice("[src] examines [bait] closer, and telekinetically shatters the pen."))
+	for(var/obj/O in view(1, src))
+		if(istype(O, /obj/item/stack/ore/diamond))
+			qdel(O)
+			src.visible_message("<span class='notice'>[src] consumes [O], and it disappears! ...At least, you think.</span>")
+		else if(istype(O, /obj/item/pen/survival))
+			qdel(O)
+			src.visible_message("<span class='notice'>[src] examines [O] closer, and telekinetically shatters the pen.</span>")
```

Comment:
```suggestion
	for(var/obj/potential_consumption in view(1, src))
		if(istype(potential_consumption, /obj/item/stack/ore/diamond))
			qdel(potential_consumption)
			visible_message(span_notice("[src] consumes [potential_consumption], and it disappears! ...At least, you think."))
		else if(istype(potential_consumption, /obj/item/pen/survival))
			qdel(potential_consumption)
			visible_message(span_notice("[src] examines [potential_consumption] closer, and telekinetically shatters the pen."))
```
* We try to name our variables a bit better.
* The span classes have been turned into defines.
* We avoid using the implied `src.` start unless it's actually informing about same-named variables.

Reply:
@Rohesie Remember to draft when making changes 😄